### ISSUE TITLE
Unable to create VRF and assign region for cloud site with mso_schema_template_vrf on NDO 4.0

### ISSUE THREAD

**Issue by marinalf** [CiscoDevNet/terraform-provider-mso#155]
### Terraform Version

* Terraform v1.2.2
* MSO v0.7.0

### NDO version

* Version: 4.0(2i)

### Cloud Network Controller

* Version 25.0(5k)

### Affected Resource(s)

* mso_schema_template_vrf
* mso_schema_site_vrf
* mso_schema_site_vrf_region

### Terraform Configuration Files

```hcl
## Define Tenant

resource "mso_tenant" "tenant" {
  name         = var.tenant.tenant_name
  display_name = var.tenant.display_name
  description  = var.tenant.description
  site_associations {
    site_id           = data.mso_site.azure_site.id
    vendor            = "azure"
    azure_access_type   = "shared"
    azure_subscription_id     = var.azure.azure_subscription_id
    azure_shared_account_id   = var.azure.azure_subscription_id
  }
}

## Define schema and template

resource "mso_schema" "schema1" {
  name          = var.schema_name
  template_name = var.template_name
  tenant_id     = mso_tenant.tenant.id
}

## Associate schema and template with cloud sites

resource "mso_schema_site" "azure_site" {
  schema_id     = mso_schema.schema1.id
  site_id       = data.mso_site.azure_site.id
  template_name = mso_schema.schema1.template_name
}

## Create VRF to be stretched to Azure

resource "mso_schema_template_vrf" "vrf1" {
  schema_id        = mso_schema.schema1.id
  template         = mso_schema.schema1.template_name
  name             = var.vrf_name
  display_name     = var.vrf_name
}

resource "mso_schema_site_vrf" "azure_site" {
  schema_id     = mso_schema.schema1.id
  template_name = mso_schema_site.azure_site.template_name
  site_id       = data.mso_site.azure_site.id
  vrf_name      = mso_schema_template_vrf.vrf1.name
}

## Define Region, CIDR and Subnets in Azure

resource "mso_schema_site_vrf_region" "azure_region" {
  schema_id          = mso_schema.schema1.id
  template_name      = mso_schema_site.azure_site.template_name
  site_id            = data.mso_site.azure_site.id
  vrf_name           = mso_schema_site_vrf.azure_site.vrf_name
  region_name        = var.azure_region_name
  vpn_gateway        = false
  hub_network_enable = true #This enables VNet Peering to Infra/Hub VNet
  hub_network = {
    name        = "default"
    tenant_name = "infra"
  }
  cidr {
    cidr_ip = var.azure_cidr_ip
    primary = true

    subnet {
      ip    = var.azure_subnet_web
    }
  }
}
```

### Debug Output

mso_schema_template_vrf.vrf1: Creating...
mso_tenant.tenant: Modifying... [id=633ede0e954de1d6b007e424]
mso_tenant.tenant: Modifications complete after 6s [id=633ede0e954de1d6b007e424]

Error: "site-local VRF vrf1 in site marina-cnc in template policy-stretch: should either have a region associated to it or should be a Hosted VRF (associated to CIDR subnet in a local VRF)"{}
   with mso_schema_template_vrf.vrf1,
   on main.tf line 42, in resource "mso_schema_template_vrf" "vrf1":
  42: resource "mso_schema_template_vrf" "vrf1" {

Full debug: https://gist.github.com/marinalf/e876f1f3a4f3ed62cbfc0c4d05902d9f

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

It worked with 0.5.0 and NDO 3.7 before. Debug claims legacy plugin SDK.

**Comment by lhercot** [comment]
We are aware of this issue and working on it. This issue is due to a change in the NDO API and it might take some time to get fixed.

**Comment by lhercot** [comment]
This issue is resolved in the latest version of the provider (v0.8.1) in conjunction with the new NDO 4.1.1i.
